pkg/defaults/defaults.go

```diff
@@ -397,9 +406,44 @@ func leasesForTest(s *api.MultiStageTestConfigurationLiteral) (ret []api.StepLea
 	return
 }
 
+// rootImageResolver creates a resolver for the root image import step. We attempt to resolve the root image and
+// the build cache. If we are able to successfully determine that the build cache is up-to-date, we import it as
+// the root image.
+func rootImageResolver(client loggingclient.LoggingClient, ctx context.Context) func(root, cache *api.ImageStreamTagReference) (*api.ImageStreamTagReference, error) {
+	return func(root, cache *api.ImageStreamTagReference) (*api.ImageStreamTagReference, error) {
+		cacheTag := &imagev1.ImageStreamTag{}
+		if err := client.Get(ctx, ctrlruntimeclient.ObjectKey{Namespace: cache.Namespace, Name: fmt.Sprintf("%s:%s", cache.Name, cache.Tag)}, cacheTag); err != nil {
+			if kapierrors.IsNotFound(err) {
+				// no build cache, use the normal root
+				return root, nil
+			}
+			return nil, fmt.Errorf("could not resolve build cache image stream tag %s: %w", cache.ISTagName(), err)
```

Comment:
Maybe we could be robust against random flakes here and just use the normal root. If the cluster is in serious trouble than something will fail later, but cache flakes should not fail a job.

Reply:
I thought about that, but I'm not sure that there's any difference between failing on a GET ISTag call for the cache or the root - both are against the same API server, so /shrug